torch/fx/symbolic_trace.py

```diff
@@ -59,25 +59,19 @@ def create_arg(self, a: Any) -> Argument:
         # tensor value into a special attribute on the Module s.t. we can
         # retrieve it with a get_attr.
         if isinstance(a, torch.Tensor):
-            # TODO: slow
-            def search_for_tensor(m : torch.nn.Module) -> Optional[List[str]]:
-                """
-                Search for a tensor value in the module's attributes. If it's
-                found, return the qualified name of that attribute, given the
-                previous `qualname_atoms`. If it's not found, recurse down into
-                child submodules. If it's not found there, return None
-                """
-                for n, p in m.__dict__.items():
-                    if a is p:
-                        return [n]
-                for n, c in m.named_children():
-                    maybe_result : Optional[List[str]] = search_for_tensor(c)
-                    if maybe_result:
-                        return [n] + maybe_result
-                return None
-            # Retrieve the qualname for an existing Tensor attribute
-            qualname_atoms : Optional[List[str]] = search_for_tensor(self.root)
-            qualname = '.'.join(qualname_atoms) if qualname_atoms else None
+            if not hasattr(self, 'tensor_attrs'):
```

Comment:
hm, checking the attr is a bit hacky, maybe at least set it to None in the constructor?

In general - why not just move it to `trace` and run proactively? it seems like a better entry point

Reply:
Yeah can move it to `trace()`